libsolidity/AST.cpp

```diff
@@ -403,7 +403,8 @@ void Assignment::checkTypeRequirements()
 	m_leftHandSide->checkTypeRequirements();
 	m_leftHandSide->requireLValue();
 	//@todo later, assignments to structs might be possible, but not to mappings
-	if (!m_leftHandSide->getType()->isValueType() && !m_leftHandSide->isLocalLValue())
+	if (m_leftHandSide->getType()->getCategory() != Type::Category::ByteArray &&
+			!m_leftHandSide->getType()->isValueType() && !m_leftHandSide->isLocalLValue())
```

Comment:
I edited this for enums too. Maybe we could add some predicate function? Like `isAssignable()` or something?